src/vcDBF.h

```diff
@@ -0,0 +1,64 @@
+#ifndef vcDBase_h__
+#define vcDBase_h__
+
+#include "udResult.h"
+#include "udFile.h"
+
+enum vcDBF_DataMappedType
+{
+  vcDBFDMT_String,
+  vcDBFDMT_Float,
+  vcDBFDMT_Logical,
+  vcDBFDMT_Integer,
+  vcDBFDMT_Memo,
+  vcDBFDMT_Invalid
+};
+
+struct vcDBF;
+
+struct vcDBF_Record
+{
+  bool deleted;
+  union vcDBF_RecordField
+  {
+    char *pString;
+    double floaty;
+    bool logical;
+    int integer;
+  } *pFields;
+};
+
+udResult vcDBF_Create(vcDBF **ppDBF);
+udResult vcDBF_Destroy(vcDBF **ppDBF);
+
+udResult vcDBF_Load(vcDBF **ppDBF, const char *pFilename);
+udResult vcDBF_Save(vcDBF *pDBF, const char *pFilename);
+
+udResult vcDBF_FindFieldIndex(vcDBF *pDBF, const char *pFieldName, uint16_t *pIndex);
+udResult vcDBF_GetFieldIndex(vcDBF *pDBF, const char *pFieldName, uint16_t *pIndex);
+uint16_t vcDBF_GetFieldCount(vcDBF *pDBF);
+uint32_t vcDBF_GetRecordCount(vcDBF *pDBF);
+
+// These two can only be performed if there's no data in pDBF
+udResult vcDBF_AddField(vcDBF *pDBF, char *pFieldName, char fieldType, uint8_t fieldLen);
+udResult vcDBF_RemoveField(vcDBF *pDBF, uint16_t fieldIndex);
+
+// Record indexes here start from 0 not 1, for SHP files need to pass in recNumber - 1
```

Comment:
If DBase files usually use an index starting from 1 it might be better to follow that system? Also, I appreciate that you're also writing the SHP file reader but this comment shouldn't contain anything to do with that. It implies that every usage of this system should also update this comment and document what index it starts at.